### PR TITLE
Update filtering & Refactor code to use Datastore API

### PR DESCRIPTION
**Pull request by cgregori**

### CODE REVIEW

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 1]
```diff
@@ -89,88 +93,104 @@ private CompositeFilter buildCompositeFilter(String zoomStr, String city, String
...
-            // TODO: log and handle error.
+            //LOGGER.log(Level.WARNING, e.getCause().getMessage());
         }
         try {
             int month = Integer.parseInt(monthStr);
-            filters.add(PropertyFilter.eq("Month", month));
+            filters.add(FilterOperator.EQUAL.of("Month", month));
         } catch (NumberFormatException e) {
-            // TODO: log and handle error.
+            //LOGGER.log(Level.WARNING, e.getCause().getMessage());
```
Comment (cgregori) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:121]: I'm not entirely sure why, but the code only runs when all the loggers are commented out in the catch blocks.

Comment (thompsond) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:121]: Oh, I encountered this earlier, sometimes the cause is null but there is still a message. So either check if the cause is null or use e.getMessage()!

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 2]
```diff
@@ -89,88 +93,104 @@ private CompositeFilter buildCompositeFilter(String zoomStr, String city, String
...
+    */
+    private MapImage entityToMapImage(Entity entity) { 
+        double latitude = (double) entity.getProperty("Latitude");
+        double longitude = (double) entity.getProperty("Longitude");
+        long zoom = (long) entity.getProperty("Zoom");
+        String cityName = (String) entity.getProperty("City Name");
+        long month = (long) entity.getProperty("Month");
+        long year = (long) entity.getProperty("Year");
+        String timeStamp = (String) entity.getProperty("Time Stamp");
+        System.out.println("H");
```
Comment (thompsond) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:191]: Was this for debugging?

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 3]
```diff
@@ -89,88 +93,104 @@ private CompositeFilter buildCompositeFilter(String zoomStr, String city, String
...
+            FilterOperator.LESS_THAN_OR_EQUAL.of("Month", monthTo),
+            FilterOperator.LESS_THAN_OR_EQUAL.of("Year", yearTo)));
     }
 
-    private ArrayList<MapImage> entitiesToMapImages(QueryResults<Entity> resultList) {
+    private ArrayList<MapImage> entitiesToMapImages(PreparedQuery resultList) {
         ArrayList<MapImage> resultMapImages = new ArrayList<>();
-        while (resultList.hasNext()) {  // While we still have data
-            resultMapImages.add(entityToMapImage(resultList.next())); // Add the MapImage to the List
+        for(Entity entity : resultList.asIterable()) {
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:172]: nit space after for

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 1 above)
Comment (theannielin) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:121]: Yup, I'd suggest doing that before merging.

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 4]
```diff
@@ -89,88 +93,104 @@ private CompositeFilter buildCompositeFilter(String zoomStr, String city, String
...
             int year = Integer.parseInt(yearStr);
-            filters.add(PropertyFilter.eq("Year", year));
+            filters.add(FilterOperator.EQUAL.of("Year", year));
         } catch (NumberFormatException e) {
-            // TODO: log and handle error.
+           // LOGGER.log(Level.WARNING, e.getCause().getMessage());
+        }
+        try {
+            // TODO: add the hotel date range UI
+            // filters.add(buildDateFilters(0, 0, 0, 0));
```
Comment (theannielin) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:131]: Is this supposed to be commented out?

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 2 above)
Comment (cgregori) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:191]: Yeah, sorry about that. Thought I got rid of all those

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 4 above)
Comment (cgregori) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:131]: Yes, the UI isn't in place for it to work yet and the feature is incomplete. I added the try catch block here to signify that the buildDateFilters should be placed here for future development.

Comment (theannielin) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:131]: Gotcha, could you elaborate on that in the above TODO?

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 1 above)
Comment (cgregori) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:121]: Sg

File: src/main/java/com/google/step/FrontendQueryDatastore.java (hunk 4 above)
Comment (cgregori) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:136]: Sg

**Pull request merged**